Mailbox/DataLoader.cs

```diff
@@ -1,21 +1,84 @@
-﻿using System.Collections.Generic;
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
 using System.IO;
 
 namespace Mailbox
 {
-    public class DataLoader
+    public class DataLoader : IDisposable
     {
+        private readonly Stream Source;
+        private bool StreamDisposed = false;
         public DataLoader(Stream source)
         {
+            if(!(source is null))
+            {
+                Source = source;
+            }
+            else
+            {
+                throw new ArgumentNullException("DataLoader's Stream cannot be null");
+            }
         }
 
         public List<Mailbox> Load()
         {
+            Source.Position = 0;
+
+            List<Mailbox> mailboxes = new List<Mailbox>();
+
+            try
+            {
+                using (StreamReader sr = new StreamReader(Source))
+                {
+                    while(!sr.EndOfStream)
```

Comment:
```suggestion
                    return JsonConvert.DeserializeObject<List<MailBox>>(reader.ReadToEnd());
```
The try block can be significantly shortened by reading the lines straight into a MailBox list and returning. It will also fix the build errors